contribs/gnodev/cmd/gnodev/main.go

```diff
@@ -366,6 +408,44 @@ func runEventLoop(
 						Error("unable to reset node state", "err", err)
 				}
 
+			case rawterm.KeyCtrlS: // Save
+				logger.WithGroup(NodeLogName).Info("saving state...")
+				if err := dnode.SaveCurrentState(ctx); err != nil {
+					logger.WithGroup(NodeLogName).
+						Error("unable to save node state", "err", err)
+				}
+
+			case rawterm.KeyE:
+				logger.WithGroup(NodeLogName).Info("exporting state...")
+				doc, err := dnode.ExportStateAsGenesis(ctx)
+				if err != nil {
+					logger.WithGroup(NodeLogName).
+						Error("unable to export node state", "err", err)
+					continue
+				}
+
+				docfile := filepath.Join(path, fmt.Sprintf("export_%d.jsonl", exported))
+				if err := doc.SaveAs(docfile); err != nil {
+					logger.WithGroup(NodeLogName).
+						Error("unable to save genesis", "err", err)
+				}
+				exported++
+
+				logger.WithGroup(NodeLogName).Info("node state exported", "file", docfile)
```

Comment:
Insane nitpick: there isn't a newline before the case as with other cases

Reply:
i like having consistent space, done: 79f2e2fec796409989324bdce39da9c01ab6f383